lib/storage/file.js

```diff
@@ -796,6 +796,145 @@ File.prototype.getSignedUrl = function(options, callback) {
+    throw new Error('An expiration date cannot be in the past.');
+  }
+
+  var expirationString = new Date(options.expiration).toISOString();
+  var conditions = []; 
+
+  // Populate eq conditions
+  if (options.equals && options.equals instanceof Array) {
+    var equalsLength = options.equals.length;
+    for (var i = 0; i < equalsLength; i++) {
+      var equalCondition = options.equals[i];
+      if ((!(equalCondition instanceof Array)) || equalCondition.length !== 2) {
+        throw new Error('Equals condition must be an array of 2 elements.');
+      } else {
+        conditions.push(['eq', equalCondition[0], equalCondition[1]]);
+      }
+    }
+  }
+
+  // Populate starts-with conditions
+  if (options.startsWith && options.startsWith instanceof Array) {
+    var startsWithLength = options.startsWith.length;
+    for (var j = 0; j < startsWithLength; j++) {
+      var startCondition = options.startsWith[j];
+      if (!(startCondition instanceof Array) || startCondition.length !== 2) {
+        throw new Error('StartsWith condition must be an array of 2 elements.');
+      } else {
+        conditions.push(['starts-with', startCondition[0], startCondition[1]]);
+      }
+    }
+  }
+
+  // Eq condition for object name
+  conditions.push(['eq', '$key', this.name]);
+
+  // The bucket that will hold the object
+  conditions.push({'bucket': this.bucket.name});
+  
+  // The ACL for the file, if any provided
+  if (options.acl) {  
+    conditions.push({'acl': options.acl});
+  }
+
+  // The redirect URL in case of successfull upload
+  if (options.successRedirect) {
+    conditions.push({'success_action_redirect': options.successRedirect});
+  }
+
+  // The response status in case of successful upload
+  if (options.successStatus) {
+    conditions.push({'success_action_status': options.successStatus});
+  }
+
+  // Range for the content length
+  if (options.contentLengthRange) {
+    var min = options.contentLengthRange.min;
+    var max = options.contentLengthRange.max;
+    if (typeof min === 'undefined' || 
+      typeof max === 'undefined') {
+      throw new Error('ContentLengthRange must have min and max fields.');
+    }
+    conditions.push(['content-length-range', min, max]);
+  }
+
+  // Policy object
+  var policy = {
+    expiration: expirationString,
+    conditions: conditions
+  };
+
+  var makeAuthorizedRequest_ = this.bucket.storage.makeAuthorizedRequest_;
+
+  makeAuthorizedRequest_.getCredentials(function(err, credentials) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var sign = crypto.createSign('RSA-SHA256');
+    var policyString = JSON.stringify(policy);
+    var policyBase64 = new Buffer(policyString).toString('base64');
+
+    sign.update(policyBase64);
+
+    var signature = sign.sign(credentials.private_key, 'base64');
+
+    callback(null, {
+      string: policyString,
+      base64: policyBase64,
+      signature: signature  
+    });
+  });
+};
```

Comment:
I think you can remove the comments you used in this method. Your architecture of the variable names and logical handling of the input is self-documenting :)